src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs

```diff
@@ -144,6 +143,18 @@ public CompilationResult Compile(RelativeFileInfo fileInfo, string compilationCo
 
                     if (!result.Success)
                     {
+                        if (!compilation.References.Any() && !_applicationReferences.Value.Any())
```

Comment:
Which is sort of what `~_applicationReferences.Value.Any()` represents. But it's a bit lossy compared to directly querying if the app is missing that flag in the project.json.